src/kixi/nhs/board_report.clj

```diff
@@ -0,0 +1,173 @@
+(ns kixi.nhs.board-report
+  (:require [kixi.nhs.data.storage :as storage]
+            [clojure.tools.logging :as log]
+            [clojure.edn           :as edn]
+            [clj-time.core         :as t]
+            [clj-time.format       :as tf]
+            [cheshire.core         :as json]
+            [kixi.ckan.data        :as data]))
+
+(def custom-formatter (tf/formatter "yyyyMMddHHmmss"))
+
+(defn now->str
+  "Formats the current timestamp into a string
+  with date and time."
+  []
+  (let [now (t/now)]
+    (tf/unparse custom-formatter now)))
+
+(defn filter-and-enrich-dataset
+  "Filters and enriches dataset according to given recipe."
+  [recipe-map data]
+  (let [{:keys [indicator-field conditions indicator-id]} recipe-map]
+    (if (contains? (first data) indicator-field)
+      (keep (fn [d] (when (every? (fn [condition] (let [{:keys [field value]} condition]
+                                                    (= (get d field) value)))
+                                  conditions)
+                      (-> d
+                          (select-keys ["Year"])
+                          (assoc "Indicator id" indicator-id "Value" (get d indicator-field)))))
+            data)
+      '())))
```

Comment:
I'll change the empty sequence to an empty vector.
We're not using 'nil' as for some cases it'll return an empty sequence and we'd have to add another condition to get 'nil'.